gateway/src/chains/ethereum/ethereum.validators.ts

```diff
@@ -0,0 +1,149 @@
+
+export const invalidPrivateKeyError: string =
+  'The privateKey param is not a valid Ethereum private key (64 hexidecimal characters).';
+
+export const invalidSpenderError: string =
+  'The spender param is not a valid Ethereum public key (0x followed by 40 hexidecimal characters).';
+
+export const invalidTokenSymbolsError: string =
+  'The tokenSymbols param should be an array of strings.';
+
+export const invalidTokenError: string = 'The token param should be a string.';
+
+export const invalidAmountError: string =
+  'If amount is included it must be a string of a non-negative integer.';
+
+export const invalidNonceError: string =
+  'If nonce is included it must be a non-negative integer.';
+
+export const invalidTxHashError: string = 'The txHash param must be a string.';
+
+// given a request, look for a key called privateKey that is an Ethereum private key
+export const validatePrivateKey = validate(
+  'privateKey',
+  invalidPrivateKeyError,
+  (val) => typeof val === 'string' && isPrivateKey(val)
+);
+
+// given a request, look for a key called spender that is 'uniswap' or an Ethereum public key
+export const validateSpender = validate(
+  'spender',
+  invalidSpenderError,
+  (val) => typeof val === 'string' && (val === 'uniswap' || isPublicKey(val))
+);
+
+// confirm that tokenSymbols is an array of strings
+export const validateTokenSymbols = (req: any): Array<string> => {
+  let errors: Array<string> = [];
+  if (req.tokenSymbols) {
+    if (Array.isArray(req.tokenSymbols)) {
+      req.tokenSymbols.forEach((symbol: any) => {
+        if (typeof symbol === 'string') {
+        } else {
+          errors.push(invalidTokenSymbolsError);
+        }
+      });
+    } else {
+      errors.push(invalidTokenSymbolsError);
+    }
+  } else {
+    errors.push(missingParameter('tokenSymbols'));
+  }
+  return errors;
+};
+
+// confirm that token is a string
+export const validateToken = validate(
+  'token',
+  invalidTokenError,
+  (val) => typeof val === 'string'
+);
+
+// if amount exists, confirm that it is a string of a natural number
+export const validateAmount = validate(
+  'amount',
+  invalidAmountError,
+  (val) => typeof val === 'string' && isNaturalNumberString(val),
+  true
+);
+
+export const validateNonce = validate(
+  'nonce',
+  invalidNonceError,
+  (val) => typeof val === 'number' && val > -1,
+  true
+);
+
+export const validateTxHash = validate(
+  'txHash',
+  invalidTxHashError,
+  (val) => typeof val === 'string'
+);
+
+// request types and corresponding validators
+
+export const validateEthereumNonceRequest = (
+  req: EthereumNonceRequest
+): void => {
+  throwErrorsIfExist(validatePrivateKey(req));
+};
+
+export const validateEthereumAllowancesRequest = (
+  req: EthereumAllowancesRequest
+): void => {
+  const errors = validatePrivateKey(req)
+    .concat(validateSpender(req))
+    .concat(validateTokenSymbols(req));
+  throwErrorsIfExist(errors);
+};
+
+export const validateEthereumBalanceRequest = (
```

Comment:
There are really two types of validator functions here - the first type are the validator "components" that output a list of error strings. The other type are the request validators which raises an error if any of the components output an error.

We should declare the function types of these two validator types, for better documentation. e.g.

```
type Validator = (req: any) => Array<string>;
type RequestValidator = (req: any) => void;
```

Also, a lot of the request validator functions are really the same - as in they're concatenating the results from multiple validators. It looks like we can just turn that into a factory method for the common case.